### PR TITLE
add service account component

### PR DESCRIPTION
**Pull request by xh3b4sd**
This PR adds a service account component and fixes the vendored dependencies. Now tests pass again. Circle does not build and succeeds anyway, but this is another story. See https://github.com/giantswarm/certificatetpr/issues/12 for tracking.

### CODE REVIEW

**Comment by xh3b4sd** [comment]
@rossf7 since you are driving the overarching story feel free to hijack and merge.

**Review by rossf7 — COMMENTED** [review]
Is the PR missing a commit? I don't see the service account component.

File: tls_assets.go [hunk 1]
```diff
@@ -1,68 +1,84 @@
...
+	// WorkerComponent is a worker component.
 	WorkerComponent ClusterComponent = "worker"
-	// EtcdComponent is the etcd cluster
+	// EtcdComponent is the etcd cluster component.
 	EtcdComponent ClusterComponent = "etcd"
-	// CalicoComponent is the calico component
+	// CalicoComponent is the calico component.
 	CalicoComponent ClusterComponent = "calico"
+	// ServiceAccountComponent is the service-account component.
+	ServiceAccountComponent ClusterComponent = "service-account"
```
Comment (xh3b4sd) [inline tls_assets.go:43]: I added the constant here.

**Comment by rossf7** [comment]
Ignore me its in the vendor commit. LGTM

File: tls_assets.go [hunk 2]
```diff
@@ -1,68 +1,84 @@
...
-// ClusterComponents is a slice enumerating all the components that make up the cluster
-var ClusterComponents = []ClusterComponent{APIComponent, WorkerComponent, EtcdComponent, CalicoComponent}
+// ClusterComponents is a slice enumerating all the components that make up the
+// cluster.
+var ClusterComponents = []ClusterComponent{
+	APIComponent,
+	WorkerComponent,
+	EtcdComponent,
+	CalicoComponent,
+	ServiceAccountComponent,
```
Comment (xh3b4sd) [inline tls_assets.go:74]: I added the constant to the components list here.

**Comment by xh3b4sd** [comment]
@rossf7 sorry I messed up the commit because of the vendoring magic.

**Review by rossf7 — APPROVED** [review]

**Pull request merged**